internal/security/proxy/service.go

```diff
@@ -0,0 +1,370 @@
+/*******************************************************************************
+ * Copyright 2019 Dell Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ * @author: Tingyu Zeng, Dell
+ * @version: 1.1.0
+ *******************************************************************************/
+package proxy
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	"github.com/dghubble/sling"
+	"github.com/edgexfoundry/edgex-go/internal/pkg/config"
+)
+
+type Service struct {
+	client Requestor
+}
+
+func NewService(r Requestor) Service {
+	return Service{
+		client: r,
+	}
+}
+
+func (s *Service) CheckProxyServiceStatus() error {
+	return s.checkServiceStatus(Configuration.KongURL.GetProxyBaseURL())
+}
+
+func (s *Service) CheckSecretServiceStatus() error {
+	return s.checkServiceStatus(Configuration.SecretService.GetSecretSvcBaseURL())
+}
+
+func (s *Service) checkServiceStatus(path string) error {
+	req, err := sling.New().Get(path).Request()
+	resp, err := s.client.Do(req)
+	if err != nil {
+		e := fmt.Sprintf("the status of service on %s is unknown, the initialization is terminated", path)
+		return errors.New(e)
+	}
+	defer resp.Body.Close()
+
+	switch resp.StatusCode {
+	case http.StatusOK:
+		LoggingClient.Info(fmt.Sprintf("the service on %s is up successfully", path))
+		break
+	default:
+		err = fmt.Errorf("unexpected http status %v %s", resp.StatusCode, path)
+		LoggingClient.Error(err.Error())
+		return err
+	}
+	return nil
+}
+
+func (s *Service) ResetProxy() error {
+	paths := []string{RoutesPath, ServicesPath, ConsumersPath, PluginsPath, CertificatesPath}
+	for _, path := range paths {
+		d, err := s.getSvcIDs(path)
+		if err != nil {
+			return err
+		}
+		for _, c := range d.Section {
+			r := &Resource{c.ID, s.client}
+			err = r.Remove(path)
+			if err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func (s *Service) Init(cert CertificateLoader) error {
+	err := s.postCert(cert)
+	if err != nil {
+		return err
+	}
+
+	services := config.ListDefaultServices()
+	//TODO: There are specific other services being routed through Kong. This should probably be driven through configuration
+	//      For now, I'm putting this inline
+	services["edgex-support-rulesengine"] = "Rulesengine"
+	services["edgex-device-virtual"] = "Virtualdevice"
```

Comment:
should we use the service name from per-defined in constant already?

Reply:
I'm going to hold off on this. These services aren't yet defined as constants. If they were, they'd be here
https://github.com/edgexfoundry/go-mod-core-contracts/blob/99c519f61f7d3ce76deed2183769c717f123c3b4/clients/constants.go#L56

Reply:
ok, probably create an open issue for this and so we won't forget this